src/main/java/net/mcreator/ui/component/JMinMaxSpinner.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.component;
+
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.init.L10N;
+
+import javax.swing.*;
+import javax.swing.event.ChangeListener;
+import java.awt.*;
+
+public class JMinMaxSpinner extends JPanel {
+
+	private final JSpinner min;
+	private final JSpinner max;
+
+	public JMinMaxSpinner(double minVal, double maxVal, double smin, double smax, double step) {

```

Comment:
I think this new class should also have a constructor to take integer arguments (because the `JSpinner` itself supports different number types, I also was already thinking to use this for number type block properties).